wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -223,6 +226,84 @@ newUnsignedTransaction ActiveWallet{..} options accountId payees = runExceptT $
     toTxOut :: (Address, Coin) -> TxOutAux
     toTxOut (a, c) = TxOutAux (TxOut a c)
 
+-- | Creates a new unsigned transaction.
+--
+-- NOTE: this function does /not/ perform a payment, it just creates a new
+-- transaction which will be signed and submitted to the blockchain later.
+-- It returns a transaction and a list of source addresses with corresponding
+-- derivation paths.
+prepareUnsignedTxWithSources :: ActiveWallet
+                             -> CoinSelectionOptions
+                             -- ^ The options describing how to tune the coin selection.
+                             -> HdAccountId
+                             -- ^ The source HD account from where the payment should originate
+                             -> NonEmpty (Address, Coin)
+                             -- ^ The payees
+                             -> PassPhrase
+                             -> IO (Either NewTransactionError
+                                           ( Tx
+                                           , NonEmpty (Address, [Word32])
```

Comment:
I still think that some form of type synonym for `[Word32]` would be nice but I won't insist 😉 

have you checked, out of curiosity, if `cardano-crypto` has already a "bip-44-like" type you can borrow? This is a minor thing though.